to-do-ui/src/app/shared/event-gateway.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable } from "@angular/core";
+import { EventEmitter } from "events";
+import { EventTypeEnum } from "./event-type.enum";
+import { EventModel } from "./event.model";
+
+@Injectable({
+	providedIn: "root"
+})
+export class EventGatewayService {
+
+	private eventEmitter = new EventEmitter();
+	private events: { [key: number]: EventModel } = {};
+	private nextSubscriptionId: number = 1;
+
+	public subscribe(eventType: EventTypeEnum, callback: (...args: any[]) => void): number {
+		const subscriptionId = this.nextSubscriptionId;
+		this.nextSubscriptionId++;
+		this.events[subscriptionId] = { name: eventType, handler: callback } as EventModel;
+		this.eventEmitter.on(eventType, callback);
+
+		return subscriptionId;
+	}
+
```

Comment:
also add unsubscribe(...)

```suggestion
public unsubscribe(subscriptionId: number): void {
		if (this.events[subscriptionId]) {
			this.eventEmitter.off(this.events[subscriptionId].name, this.events[subscriptionId].handler);
			delete this.events[subscriptionId];
		} else {
			console.log(`[EventGatewayService] Event with id ${subscriptionId} not found!`);
		}
	}
```